### DISCUSSION TITLE
How to display citations in a response instead of displaying all citations directly below the response?

### DISCUSSION THREAD

**Discussion by secsilm** [langgenius/dify#10897] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I am using dify for RAG on an internal knowledge base, and I would like to cite sources when answering questions. Then I found the following two issues:

1. Displaying too many citations. There are 10 citations shown below the reply, but it is clear that only 3 are used in the reply. How can I display only the 3 cited?
2. Displaying all citations below the response. The response lists 3 points, and I want to display the referenced sources directly after each point, just like the latest ChatGPT, instead of showing them all below the response. How can I achieve this?
![image](https://github.com/user-attachments/assets/8727fa14-05b1-4b42-912a-5ab7ba8eefda)

**Comment by efreykongcn** [comment]
I dont' think Dify has the capability to do so. This should be a new feature. Ragflow has already supported this feature.

**Reply by alliecatowo** [reply]
agreed. ragflows rag is alot more robust in some ways (obviously 😂), but its not nearly as polished as dify.  

I swear I had inline citations working once for dify, but never again. According to their docs, they do officially support inline attribution, idk what's wrong. I've used every gemini model and no dice. Maybe its only openai models right now, 4o is shown in the docs[here](https://docs.dify.ai/guides/knowledge-base/retrieval-test-and-citation). 

Definitely let me know if anyone gets dify inline attribution working with gemini models! 

<img width="400" alt="image" src="https://github.com/user-attachments/assets/66d04934-520c-425f-9c28-6d777efde6ec" />

**Reply by AaronMKk** [reply]
Hi Coleman, would you mind sharing your dsl file? Cause i do not know how to use the official citation feature

**Comment by rajihahr** [comment]
any updates on this?
